pkg/template/nstemplatetier_test.go

```diff
@@ -0,0 +1,169 @@
+					Type:     "cicd",
+				},
+				{
+					Revision: "1a2b3c",
+					Template: "{baz1}",
+					Type:     "stage",
+				},
+			},
+		},
+	}
+	advancedTier := &toolchainv1alpha1.NSTemplateTier{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "toolchain.dev.openshift.com/v1alpha1",
+			Kind:       "NSTemplateTier",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "advanced",
+			Namespace: "toolchain-host-operator",
+		},
+		Spec: toolchainv1alpha1.NSTemplateTierSpec{
+			Namespaces: []toolchainv1alpha1.Namespace{
+				{
+					Revision: "ghijkl",
+					Template: "{foo1}",
+					Type:     "ide",
+				},
+				{
+					Revision: "789012",
+					Template: "{bar1}",
+					Type:     "cicd",
+				},
+				{
+					Revision: "4d5e6f",
+					Template: "{baz1}",
+					Type:     "stage",
+				},
+			},
+		},
+	}
+	cl := fake.NewFakeClient(basicTier, advancedTier)
+
+	t.Run("success", func(t *testing.T) {
+		// given
+		hostCluster := newHostCluster(cl, fedv1b1.ClusterCondition{
+			Type:   fedcommon.ClusterReady,
+			Status: apiv1.ConditionTrue,
+		})
+		// when
+		tmpls, err := template.GetNSTemplates(hostCluster, "basic")
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, tmpls, 3)
+		assert.Equal(t, template.RevisionedTemplate{
+			Revision: "abcdef",
+			Template: "{foo1}",
+		}, tmpls["ide"])
+		assert.Equal(t, template.RevisionedTemplate{
+			Revision: "123456",
+			Template: "{bar1}",
+		}, tmpls["cicd"])
+		assert.Equal(t, template.RevisionedTemplate{
+			Revision: "1a2b3c",
+			Template: "{baz1}",
+		}, tmpls["stage"])
+	})
+
+	t.Run("failures", func(t *testing.T) {
+
+		t.Run("host cluster not available", func(t *testing.T) {
+			// given
+			hostCluster := func() (*cluster.FedCluster, bool) {
+				return nil, false
+			}
+			// when
+			_, err := template.GetNSTemplates(hostCluster, "unknown")
+			// then
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "unable to connect to the host cluster: unknown cluster")
+		})
+
+		t.Run("host cluster not ready", func(t *testing.T) {
+			// given
+			hostCluster := newHostCluster(cl, fedv1b1.ClusterCondition{
+				Type:   fedcommon.ClusterReady,
+				Status: apiv1.ConditionFalse,
+			})
+			// when
+			_, err := template.GetNSTemplates(hostCluster, "unknown")
+			// then
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "the host cluster is not ready")
+		})
+
+		t.Run("unknown tier", func(t *testing.T) {
+			// given
+			hostCluster := newHostCluster(cl, fedv1b1.ClusterCondition{
+				Type:   fedcommon.ClusterReady,
+				Status: apiv1.ConditionTrue,
+			})
+			// when
+			_, err := template.GetNSTemplates(hostCluster, "unknown")
+			// then
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "unable to retrieve the NSTemplateTier 'unknown' from 'Host' cluster")
+		})
+	})
+
+}
+
+func newHostCluster(cl client.Client, condition fedv1b1.ClusterCondition) cluster.GetHostClusterFunc {
+	return func() (*cluster.FedCluster, bool) {
+		return &cluster.FedCluster{
```

Comment:
You should have also set the `OperatorNamespace`, otherwise, when you make the call https://github.com/codeready-toolchain/member-operator/pull/63/files#diff-88f2951b69e27daa0728c455506a942aR38 the client will look for it cluster-wide, but you should look for it only inside of the namespace